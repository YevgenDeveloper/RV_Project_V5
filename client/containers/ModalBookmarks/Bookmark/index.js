// @flow

import type { BookmarkType } from 'client/types';

import React from 'react';

import { googleStorageUrl } from 'client/constants';
import styles from './styles.css';

type Props = BookmarkType & {
  onClick: void => void,
};

const Bookmark = (props: Props) => {
  const title = `Ride: ${props.ride}, camera: ${props.camera}`;

  return (
    <button
      type="button"
      className={styles.bookmark}
      onClick={() => props.onClick()}
    >
      <img
        alt={title}
        className={styles.image}
        src={`${googleStorageUrl}/${props.ride}/${props.camera}/${props.image}`}
      />
      <div>
        <div className={styles.info}>
          <b>Ride:</b>
          {` ${props.ride}`}
        </div>
        <div className={styles.info}>
          <b>Camera:</b>
          {` ${props.camera}`}
        </div>
      </div>
    </button>
  );
};

export default Bookmark;
