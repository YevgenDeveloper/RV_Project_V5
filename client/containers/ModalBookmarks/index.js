// @flow

import type { BookmarkType } from 'client/types';

import React from 'react';

import { Modal } from 'client/components';
import Bookmark from './Bookmark';

type Props = {
  images: ?{
    [cameraId: string]: string[],
  },
  bookmarks: ?(BookmarkType[]),
  close: void => void,
  onClickNavigate: BookmarkType => void,
};

const ModalBookmarks = (props: Props) => (
  <Modal close={props.close}>
    <h1>
      {props.bookmarks && props.bookmarks.length > 0
        ? 'Bookmarks'
        : 'No bookmarks'}
    </h1>
    {props.bookmarks &&
      props.bookmarks.map(({ ride, camera, image }) => (
        <Bookmark
          key={`${ride}${camera}${image}`}
          ride={ride}
          camera={camera}
          // $FlowFixMe
          image={props.images[camera][image]}
          onClick={() => props.onClickNavigate({ ride, camera, image })}
        />
      ))}
  </Modal>
);

export default ModalBookmarks;
