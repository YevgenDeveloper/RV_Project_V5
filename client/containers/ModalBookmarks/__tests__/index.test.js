// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import ModalBookmarks from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(
    <ModalBookmarks onClickNavigate={jest.fn()} close={jest.fn()} />,
  );

  expect(wrapper).toMatchSnapshot();
});
