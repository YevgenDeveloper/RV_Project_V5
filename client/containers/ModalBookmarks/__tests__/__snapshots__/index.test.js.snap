// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalBookmarks
    close={[MockFunction]}
    onClickNavigate={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h1>
          No bookmarks
        </h1>,
        undefined,
      ],
      "className": undefined,
      "close": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No bookmarks",
        },
        "ref": null,
        "rendered": "No bookmarks",
        "type": "h1",
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h1>
            No bookmarks
          </h1>,
          undefined,
        ],
        "className": undefined,
        "close": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No bookmarks",
          },
          "ref": null,
          "rendered": "No bookmarks",
          "type": "h1",
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
